src/sync-strings.py

```diff
@@ -1,108 +1,86 @@
-import git, tomlkit
 
+def getAllFilesToml(toml_path, search_path):
+    """
+    Extract a list of all files referenced in the project configuration file
+    (TOML). Files are stored with path relative to `search_path`.
+    """
 
-def ts():
-    return str(datetime.datetime.now())
+    basedir = os.path.dirname(toml_path)
+    project_config = paths.TOMLParser().parse(toml_path, env={"l10n_base": ""})
+    basedir = os.path.join(basedir, project_config.root)
 
+    print(f"Analyzing files in {toml_path}")
+    toml_files = []
 
-def log(s):
-    print(f"{ts()} {s}")
+    files = paths.ProjectFiles(None, [project_config])
+    for l10n_file, reference_file, _, _ in files:
+        toml_files.append(os.path.relpath(reference_file, search_path))
 
+    for locale in project_config.all_locales:
+        files = paths.ProjectFiles(locale, [project_config])
+        for l10n_file, reference_file, _, _ in files:
+            # Ignore missing files for locale
+            if not os.path.exists(l10n_file):
+                continue
 
-def android_locale(locale):
-    """Convert the given Pontoon locale to the code Android uses"""
-    ANDROID_COMPATIBILITY_MAPPINGS = {"he": "iw", "yi": "ji", "id": "in"}
-    if locale in ANDROID_COMPATIBILITY_MAPPINGS:
-        return ANDROID_COMPATIBILITY_MAPPINGS[locale]
-    if matches := re.match(r"([a-z]+)-([A-Z]+)", locale):
-        return f"{matches.group(1)}-r{matches.group(2)}"
-    return locale
+            # Ignore file is the reference is missing
+            if not os.path.exists(reference_file):
+                continue
 
+            toml_files.append(os.path.relpath(l10n_file, search_path))
+    toml_files.sort()
 
-def release_locales(repo_path, project_path):
-    """Load the list of locales from l10n-release.toml or from l10n.toml as fallback."""
-    for filename in ("l10n-release.toml", "l10n.toml"):
-        toml_path = os.path.join(repo_path, project_path, filename)
-        if os.path.exists(toml_path):
-            with open(toml_path) as toml_fp:
-                l10n_toml = tomlkit.loads(toml_fp.read())
-                log(f"Returning strings from {toml_path}")
-                return l10n_toml["locales"]
+    return toml_files
 
 
-def all_strings_xml_paths(repo_path, project_path, locales):
-    """Yield all combinations of strings.xml paths and locales. Returns a relative path."""
-    with open(os.path.join(repo_path, project_path, "l10n.toml")) as main_toml_fp:
-        main_l10n_toml = tomlkit.loads(main_toml_fp.read())
-        for locale in locales:
-            pathname = main_l10n_toml["paths"][0]["l10n"].replace("{android_locale}", android_locale(locale))
-            for strings_xml_path in glob.glob(os.path.join(repo_path, project_path, pathname), recursive=True):
-                yield os.path.relpath(strings_xml_path, repo_path)
+def main():
+    src_repo_path = os.environ.get("INPUT_SRC")
+    if src_repo_path is None:
+        sys.exit("Empty 'src' parameter. Exiting.")
```

Comment:
Is this an appropriate exit code? Reading the docs https://docs.github.com/en/actions/creating-actions/setting-exit-codes-for-actions indicates that it needs to be any integer but 0. So, I think it's best if we kept with the existing `exit(1)` and `print(string)`

Reply:
I believe it is correct ([sys.exit docs](https://docs.python.org/3/library/sys.html#sys.exit))

> If another type of object is passed, None is equivalent to passing zero, and any other object is printed to stderr and results in an exit code of 1. In particular, sys.exit("some error message") is a quick way to exit a program when an error occurs.

Let me know if you still want to change it.

Reply:
Side note: I've been using this approach without issues in GitHub workflows.